### PR TITLE
Issue18 checkpoints

### PR DESCRIPTION
**Pull request by dvonthenen**
### Issue 18 - refactor mount list file IO to use Mesos checkpoint code/template:

After comparing the output of the current file IO process to the checkpoint
method, the output is identical and behaves as expected. Tested multiple
scenarios in init and recovery.

Removed temporary exploratory code that compares the current json file
and replaced with the checkpoint implementation.

Please note that code has been copied from the internal mesos namespace. This
code will be deleted once the checkpoint process is exposed publically.
This code has been annotated with comments in code.

### CODE REVIEW

File: isolator/isolator/docker_volume_driver_isolator.cpp [hunk 1]
```diff
@@ -69,6 +69,145 @@ const char DockerVolumeDriverIsolatorProcess::prohibitedchars[NUM_PROHIBITED]  =
...
+std::string DockerVolumeDriverIsolatorProcess::mesosWorkingDir;
+
+//TODO temporary until checkpoints are public by mesosphere dev
+#include <stout/path.hpp>
+#include <slave/paths.hpp>
+#include <slave/state.hpp>
+using namespace mesos::internal::slave::paths;
+using namespace mesos::internal::slave::state;
+
+namespace mycheckpoint {
```
Comment (cantbewong) [inline isolator/isolator/docker_volume_driver_isolator.cpp]: how about dvdicheckpoint as a namespace?

File: isolator/isolator/docker_volume_driver_isolator.cpp [hunk 2]
```diff
@@ -69,6 +69,145 @@ const char DockerVolumeDriverIsolatorProcess::prohibitedchars[NUM_PROHIBITED]  =
...
+
+	namespace internal {
```
Comment (cantbewong) [inline isolator/isolator/docker_volume_driver_isolator.cpp]: since this is wrapped in our own namespace, do we need another nested namespace here?

Comment (cantbewong) [inline isolator/isolator/docker_volume_driver_isolator.cpp]: Messo code does not seem to indent after a namespace, also indent when used is two chars, except for function params which is 4. https://mesos.apache.org/documentation/latest/c++-style-guide/

File: isolator/isolator/docker_volume_driver_isolator.cpp (hunk 1 above)
Comment (dvonthenen) [inline isolator/isolator/docker_volume_driver_isolator.cpp]: Will include in an update

File: isolator/isolator/docker_volume_driver_isolator.cpp (hunk 2 above)
Comment (dvonthenen) [inline isolator/isolator/docker_volume_driver_isolator.cpp]: Changed the namespace to dvdicheckpoint.
Removed the nested "internal" namespace; this was just a straight copy and paste from the mesos code.
Replaced tabs with 2 spaces. Tried to bring code up to date with Mesos coding standards.

Will have in a future commit.

**Comment by dvonthenen** [comment]
Forgot to mention that I was not able to place the temporary namespace into the hpp file because of ordering conflicts with other includes. This is only temporary code and will be removed shortly.

**Pull request merged**